plugins.d/alarm-notify.sh

```diff
@@ -433,268 +391,43 @@ filter_recipient_by_criticality() {
-    for r in ${a//,/ }
-    do
-        [ "${r}" != "disabled" ] && filter_recipient_by_criticality msteam "${r}" && arr_msteam[${r/|*/}]="1"
-    done
-
-    # rocketchat
-    a="${role_recipients_rocketchat[${x}]}"
-    [ -z "${a}" ] && a="${DEFAULT_RECIPIENT_ROCKETCHAT}"
-    for r in ${a//,/ }
-    do
-        [ "${r}" != "disabled" ] && filter_recipient_by_criticality rocketchat "${r}" && arr_rocketchat[${r/|*/}]="1"
-    done
-
-    # alerta
-    a="${role_recipients_alerta[${x}]}"
-    [ -z "${a}" ] && a="${DEFAULT_RECIPIENT_ALERTA}"
-    for r in ${a//,/ }
-    do
-        [ "${r}" != "disabled" ] && filter_recipient_by_criticality alerta "${r}" && arr_alerta[${r/|*/}]="1"
-    done
-
-    # flock
-    a="${role_recipients_flock[${x}]}"
-    [ -z "${a}" ] && a="${DEFAULT_RECIPIENT_FLOCK}"
-    for r in ${a//,/ }
-    do
-        [ "${r}" != "disabled" ] && filter_recipient_by_criticality flock "${r}" && arr_flock[${r/|*/}]="1"
-    done
-
-    # discord
-    a="${role_recipients_discord[${x}]}"
-    [ -z "${a}" ] && a="${DEFAULT_RECIPIENT_DISCORD}"
-    for r in ${a//,/ }
-    do
-        [ "${r}" != "disabled" ] && filter_recipient_by_criticality discord "${r}" && arr_discord[${r/|*/}]="1"
-    done
-
-    # pagerduty.com
-    a="${role_recipients_pd[${x}]}"
-    [ -z "${a}" ] && a="${DEFAULT_RECIPIENT_PD}"
-    for r in ${a//,/ }
-    do
-        [ "${r}" != "disabled" ] && filter_recipient_by_criticality pd "${r}" && arr_pd[${r/|*/}]="1"
-    done
-
-    # fleep.io
-    a="${role_recipients_fleep[${x}]}"
-    [ -z "${a}" ] && a="${DEFAULT_RECIPIENT_FLEEP}"
-    for r in ${a//,/ }
-    do
-        [ "${r}" != "disabled" ] && filter_recipient_by_criticality fleep "${r}" && arr_fleep[${r/|*/}]="1"
-    done
-
-    # irc
-    a="${role_recipients_irc[${x}]}"
-    [ -z "${a}" ] && a="${DEFAULT_RECIPIENT_IRC}"
-    for r in ${a//,/ }
-    do
-        [ "${r}" != "disabled" ] && filter_recipient_by_criticality irc "${r}" && arr_irc[${r/|*/}]="1"
-    done
-
-    # syslog
-    a="${role_recipients_syslog[${x}]}"
-    [ -z "${a}" ] && a="${DEFAULT_RECIPIENT_SYSLOG}"
-    for r in ${a//,/ }
-    do
-        [ "${r}" != "disabled" ] && filter_recipient_by_criticality syslog "${r}" && arr_syslog[${r/|*/}]="1"
-    done
-
-    # custom
-    a="${role_recipients_custom[${x}]}"
-    [ -z "${a}" ] && a="${DEFAULT_RECIPIENT_CUSTOM}"
-    for r in ${a//,/ }
-    do
-        [ "${r}" != "disabled" ] && filter_recipient_by_criticality custom "${r}" && arr_custom[${r/|*/}]="1"
+for method_name in ${method_names} ; do
+    declare -A arr=()
+
+    for x in ${roles//,/ } ; do
+        # the roles 'silent' and 'disabled' mean:
+        # don't send a notification for this role
+        [ "${x}" = "silent" -o "${x}" = "disabled" ] && continue
+
+        role_recipients="role_recipients_${method_name}"
+        role_recipients=${!role_recipients}
+        default_recipient="DEFAULT_RECIPIENT_$(echo ${method_name} | awk '{print toupper($0)}')"
```

Comment:
```sh
default_recipient="DEFAULT_RECIPIENT_${method_name^^}"
```
does the same

Reply:
That also only works in Bash 4.  Looking again at the rest of the script though, we're already doing a number of things that only work in Bash 4, so I'll just update things to do this.